src/devices/device.ts

```diff
@@ -82,16 +97,102 @@ export class Device extends EventEmitter {
         this.hub.send(data, characteristic, callback);
     }
 
-    public subscribe (mode: number) {
+    protected sendWithMode (modeName: string, data: Buffer, characteristic: string = Consts.BLECharacteristic.LPF2_ALL, callback?: () => void) {
+        const previousMode = this.mode;
+        this.subscribe(modeName);
+        this.send(data, characteristic, () => {
+            if (previousMode) {
+                this.subscribe(previousMode);
+                if (callback) {
+                    callback();
+                }
+            }
+        });
+    }
+
+    protected sendLinearPowerCommand (value: number) {
+        return new Promise((resolve) => {
+            if (this.isWeDo2SmartHub) {
+                const data = Buffer.from([this.portId, 0x01, 0x02, value]);
+                this.send(data, Consts.BLECharacteristic.WEDO2_MOTOR_VALUE_WRITE);
+            } else {
+                const data = Buffer.from([0x81, this.portId, 0x11, 0x51, 0x00, value]);
+                this.send(data);
+            }
+            return resolve();
+        });
+    }
+
+    public subscribe (modeName: string) {
         this._ensureConnected();
-        if (mode !== this._mode) {
-            this._mode = mode;
-            this.hub.subscribe(this.portId, this.type, mode);
+        if (modeName !== this._mode) {
+            this._mode = modeName;
+
+            const modeNum = this._modes[modeName].num[this.hub.type];
+            if (modeNum === undefined) {
+                // TODO : error handling -> unsupported mode
+                return;
+            }
+            this.hub.subscribe(this.portId, this.type, modeNum);
         }
     }
 
     public receive (message: Buffer) {
-        this.emit("receive", message);
+        if (this._mode === undefined) {
+            // TODO : error handling -> no mode defined
+            return;
+        }
+
+        const mode = this._modes[this._mode];
+
+        if (!mode.values) {
+            // TODO : error handling -> no parsing informations
+            return;
+        }
+
+        const data = [];
+
+        for (let index = 0; index <= message.length; index += Consts.ValueBits[mode.values.type]) {
+            switch (mode.values.type) {
+                case Consts.ValueType.UInt8: {
+                    data.push(message.readUInt8(index));
```

Comment:
Should the raw value be changed to take into account min/max value at this point?